lib/compress.js

```diff
@@ -3112,6 +3056,63 @@ merge(Compressor.prototype, {
                 }
             }
         }
+        if (compressor.option("unsafe_Func")
+            && exp instanceof AST_SymbolRef
+            && exp.undeclared()
+            && exp.name == "Function") {
+            // new Function() => function(){}
+            if (self.args.length == 0) return make_node(AST_Function, self, {
+                argnames: [],
+                body: []
+            });
+            if (all(self.args, function(x) {
+                return x instanceof AST_String;
+            })) {
+                // quite a corner-case, but we can handle it:
+                //   https://github.com/mishoo/UglifyJS2/issues/203
+                // if the code argument is a constant, then we can minify it.
+                try {
+                    var code = "NaN(function(" + self.args.slice(0, -1).map(function(arg) {
```

Comment:
The older format, .e.g. `functoin(a){return a}()` would suffer from `a` being reduced to `undefined` and causing the whole function to be eliminated.

OTOH `NaN(function(a){return a})` would not suffer the same fate, and I pick `NaN` to avoid namespace pollution (and it's short).